tutorial/chapter/4/index.md

```diff
@@ -15,14 +13,13 @@ Here are the operators in order, from Squiggle:
 - `*`, `/`
 - `-` (unary prefix)
 
+## Operators in general
+
+Few operators in JavaScript will ever throw exceptions, but Squiggle operators throws lots of them. The reasoning behind this is Squiggle avoids implicit type coercsions, such as considering the number `0` to be equivalent to `false` in a boolean context, or `""` to be equivalent to `false`. It also doesn't consider `"34"` to be equivalent to `34` in a numeric context. If you want this behavior, you can import the global functions `Number`, `String`, or `Boolean` and explicitly wrap your data in them, like `Number("34") + 3` or `Boolean(someNumber) and foo()`. This helps catch errors where you pass the wrong data type to a function, or where in your application `""` might be a perfectly valid string you want to use, not something "falsey".
+
 ## Logical operators
 
-The logical operators (`and` and `or`) work like their JavaScript counterparts
-(`&&` and `||`). They short-circuit just like in JavaScript, but if they
-evaluate a non-boolean value on either side, they throw an exception. That means
-you can't do `x || defaultValue`. But that is an error prone construct anyway:
-take for example `0 || 123`. You probably only wanted 123 if x was undefined,
-not if it was 0.
+The logical operators `and` and `or` operate on boolean values and return boolean values. They short circuit, meaning that `false and foo()` will not evaluate `foo()`, and `true or bar()` will not evaluate `bar()`.
```

Comment:
This is purely subjective, but why limit these operators to booleans? I get that you’re trying to make things more fail-fast, but short-circuiting `and`/`or` are well-known with truthy/falsy logic, and they’re *very* useful in many situations (as long as the truthiness semantics are reasonable).